packages/core/src/tracing/instrumentation/database/prisma.js

```diff
@@ -0,0 +1,204 @@
+                return;
+              }
+
+              // We attach the provider and destination URL to the Prisma client instance. That way, when multiple
+              // Prisma client's are used, we do not confuse providers/database URLs
+              this.__inProvider = activeDatasource.activeProvider;
+              const dataSourceUrlObject = activeDatasource.url;
+              if (dataSourceUrlObject && dataSourceUrlObject.value) {
+                this.in_DataSourceUrl = redactPassword(activeDatasource.activeProvider, dataSourceUrlObject.value);
+              } else if (dataSourceUrlObject && dataSourceUrlObject.fromEnvVar) {
+                this.in_DataSourceUrl = redactPassword(
+                  activeDatasource.activeProvider,
+                  process.env[dataSourceUrlObject.fromEnvVar]
+                );
+              }
+            });
+          } else {
+            logger.debug(
+              'The operation getConfig did not yield a thenable, the database URL will not be captured on spans.'
+            );
+          }
+        } else {
+          logger.debug(
+            'PrismaClient#_engine.getConfig does not exist, the database URL will not be captured on spans.'
+          );
+        }
+      }
+    }
+    prismaClientModule.PrismaClient = InstanaPrismaClient;
+  }
+}
+
+function shimRequest(prismaClientModule) {
+  if (
+    !prismaClientModule ||
+    !prismaClientModule.PrismaClient ||
+    !prismaClientModule.PrismaClient.prototype ||
+    !prismaClientModule.PrismaClient.prototype._request
+  ) {
+    logger.debug('prismaClientModule.PrismaClient.prototype._request does not exist, will not instrument Prisma.');
+    return;
+  }
+
+  // The relevant source file is:
+  // https://github.com/prisma/prisma/blob/main/packages/client/src/runtime/getPrismaClient.ts
+  shimmer.wrap(prismaClientModule.PrismaClient.prototype, '_request', instrumentRequest);
+}
+
+function instrumentRequest(original) {
+  return function () {
+    if (cls.skipExitTracing({ isActive })) {
+      return original.apply(this, arguments);
+    }
+
+    const argsForOriginalRequest = new Array(arguments.length);
+    for (let i = 0; i < arguments.length; i++) {
+      argsForOriginalRequest[i] = arguments[i];
+    }
+    return instrumentedRequest(this, original, argsForOriginalRequest);
+  };
+}
+
+function instrumentedRequest(ctx, originalRequest, argsForOriginalRequest) {
+  return cls.ns.runAndReturn(() => {
+    const span = cls.startSpan('prisma', EXIT);
+    span.stack = getStackTrace(instrumentedRequest, 1);
+    const params = argsForOriginalRequest[0] || {};
+    span.data.prisma = {
+      model: params.model,
+      action: params.action,
+      provider: ctx.__inProvider,
+      url: ctx.in_DataSourceUrl
+    };
+    const requestPromise = originalRequest.apply(ctx, argsForOriginalRequest);
+    if (!requestPromise && typeof requestPromise.then !== 'function') {
+      span.cancel();
```

Comment:
_Question_: Wondering why do we even start the span? We could call the original fn and start the span if we get a promise back. Ignore if you want to :)

Reply:
I think that would be "less correct", strictly speaking? We want the prisma exit span to be the active span while `originalRequest.apply` is running. In practice it probably makes little difference, because we do not expect further activity that we would want to trace while that is happening. But one case where it could make a difference is our usual check that we do not start another exit span when an exit span is already active. For that to work we want the prisma exit to be active for the duration of `_request`. WDYT?